### PR TITLE
Add workflow to sync dashboard crds automatically

### PR DESCRIPTION
**Pull request by maroroman**
Resolves: #200 
Prerequisite: https://github.com/opendatahub-io/odh-dashboard/pull/996

## Description
This PR awaits a dispatch from another workflow (https://github.com/maroroman/odh-dashboard/blob/239d23cd54addf29b88789d7a0b38f9a4dcc6313/.github/workflows/dispatch_on_crd_change.yaml) in the odh-dashboard repo and automatically creates a PR with these changes.

## How Has This Been Tested?
This PR was tested using https://github.com/nektos/act to locally verify that the workflow is sending the dispatch correctly.

To test this in real conditions, a fork of the operator and dashboard is required:

1. Create a pull request in the odh-dashboard fork that changes a file in /manifests/crd folder
2. Merge the PR into the main branch of the fork
3. Verify that the action has been triggered
4. Verify dispatch reached opendatahub-operator fork sync workflow and has triggered it.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/sync_dashboard_crds.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Git checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+      - name: Gather files
+        shell: bash
+        run: |
+          cd ${{ github.workspace }}/config/crd/dashboard-crds
+          svn checkout --force https://github.com/opendatahub-io/odh-dashboard/branches/main/manifests/crd
+          svn revert -R ./crd
```
Comment (VaishnaviHire) [inline .github/workflows/sync_dashboard_crds.yaml:21]: @maroroman Not exactly sure on how the `svn revert` command working here? Is it reverting any uncommited changes?

Can we instead just remove the existing version of crds and replace them with `odh-dashboard/branches/main/manifests/crd` ?

Comment (maroroman) [reply .github/workflows/sync_dashboard_crds.yaml:21]: That is what the command should do, as far as I know, when I looked it up and tested it the revert command with the -R argument "reverts" the state of the folder to whatever has been checked out. So this basically overwrites whatever is in the crd folder with the crd changes in the dashboard.

Comment (maroroman) [reply .github/workflows/sync_dashboard_crds.yaml]: Okay I replaced the checkout and revert command with a better `svn export` command. I have verified that it works locally. Could you please retry your test @VaishnaviHire?

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**